### PR TITLE
14 setup week and day entities

### PR DESCRIPTION
**Pull request by DanielLivshits**

### CODE REVIEW

File: src/main/java/entities/Week.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class Week{
+    private final ArrayList<Day> days;
+
+    public Week(ArrayList<Day> days){
+        this.days = days;
+    }
+
```
Comment (missionStrawberry) [inline src/main/java/entities/Week.java:12]: add a default constructor that accomplishes the same thing as the existing constructor but with no parameters

File: src/test/java/entities/DayTest.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+public class DayTest {
+
+    @Test
+    public void testEmptyConstructor(){
+        Day d = new Day(new HashMap<>(), new HashMap<>());
+        assertEquals(new HashMap<>(), d.getEvents());
+        assertEquals(new HashMap<>(), d.getTasks());
+    }
+}
```
Comment (missionStrawberry) [inline src/test/java/entities/DayTest.java:17]: Would benefit from more thorough testing once Event and Task classes are created, but with the available code at time of writing the testing is as extensive as possible.

File: src/test/java/entities/WeekTest.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class WeekTest {
+    @Test
+    public void TestEmptyConstructor(){
```
Comment (missionStrawberry) [inline src/test/java/entities/WeekTest.java:12]: this test name is a little confusing

**Review by missionStrawberry — CHANGES_REQUESTED** [review]
Left a few comments on things that can be improved, and documentation might need to be added once everyone establishes our conventions and we ask about what's required of us for documentation.

File: src/test/java/entities/DayTest.java (hunk 2 above)
Comment (Fard-Faru) [reply src/test/java/entities/DayTest.java:17]: This naming convention is not consistent with your WeekTest.java test name

**Comment by Fard-Faru** [comment]
Ill review again after Anna's comments are resolved but we as well need to add documentation

**Comment by DanielLivshits** [comment]
Fixed default constructors and test names. More tests are needed but cannot be written without implementation for the Task and Event classes. Holding off on writing documentation until conventions are agreed upon.

**Review by Fard-Faru — APPROVED** [review]
Looks Good

**Review by scissorkhizer — APPROVED** [review]
Everything looks good. Thanks for setting up the entities and the template we should follow for documentation.

**Review by POPPININ — CHANGES_REQUESTED** [review]
Having two constructors in the Day and Week classes seems unnecessary, and leaves room for code smells related to "Speculative Generality". We should write code which will allow us to have data persistence from day 1.

**Review by POPPININ — DISMISSED** [review]
Looks good!

File: src/main/java/entities/TaskFactory.java [hunk 4]
```diff
@@ -1,4 +1,20 @@
...
+ * Last Modified: NA
+ */
 package entities;
 
 public class TaskFactory {
+    /**
+     * Creates a new Task
+     * Precondition: title is a non-empty string
+     * @param title title of the task
+     * @return a task create with the given inputs
```
Comment (missionStrawberry) [inline src/main/java/entities/TaskFactory.java:15]: minor typo in docstring. created not create

File: src/test/java/entities/DayFactoryTest.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        assertEquals(d.getTasks(), d1.getTasks());
+        assertEquals(d.getEvents(), d1.getEvents());
+    }
+
+    @Test
+    public void testCreateDayWithParameters(){
+        Day d = new Day(new HashMap<>(), new HashMap<>());
+        DayFactory fact = new DayFactory();
+        Day d1 = fact.createDay(new HashMap<>(), new HashMap<>());
+        assertEquals(d.getTasks(), d1.getTasks());
```
[the remark this answers is not in the document]
Comment (DanielLivshits) [reply src/test/java/entities/DayFactoryTest.java:42]: A significant oversight on my part. Thank you for pointing it out.

[the remark this answers is not in the document]
Comment (missionStrawberry) [reply src/test/java/entities/DayFactoryTest.java]: No problem :)

**Review by missionStrawberry — DISMISSED** [review]
Looks good now!

**Review by POPPININ — DISMISSED** [review]
Fix variable names in test cases.

**Review by Fard-Faru — APPROVED** [review]

**Review by scissorkhizer — DISMISSED** [review]

File: src/test/java/entities/DayTest.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    public static void setUpHashMaps(){
+        eventMap.put("1", event1);
+        eventMap.put("2", event2);
+        taskMap.put("Hello,", task1);
+        taskMap.put("World!", task2);
+    }
+
+    @Test
+    public void testGetEvents(){
+        Day d = new Day(taskMap, eventMap);
```
Comment (missionStrawberry) [inline src/test/java/entities/DayTest.java:42]: missed a few variables for renaming in this test file

**Review by missionStrawberry — CHANGES_REQUESTED** [review]

**Review by missionStrawberry — APPROVED** [review]
Looks good now!

**Review by Fard-Faru — APPROVED** [review]

**Review by scissorkhizer — APPROVED** [review]

**Pull request merged**